blockdag/dagio.go

```diff
@@ -302,85 +304,84 @@ func (dag *BlockDAG) initDAGState() error {
 	}
 
 	// Attempt to load the DAG state from the database.
-	return dag.db.View(func(dbTx database.Tx) error {
-		// Initialize the reachability store
-		log.Infof("Loading reachability data...")
-		err = dag.reachabilityStore.init(dbaccess.NoTx())
+
+	// Initialize the reachability store
+	log.Infof("Loading reachability data...")
+	err = dag.reachabilityStore.init(dbTx)
+	if err != nil {
+		return err
+	}
+
+	// Initialize the multiset store
+	log.Infof("Loading multiset data...")
+	err = dag.multisetStore.init(dbTx)
+	if err != nil {
+		return err
+	}
+
+	// Apply the loaded utxoCollection to the virtual block.
+	dag.virtual.utxoSet, err = newFullUTXOSetFromUTXOCollection(fullUTXOCollection)
+	if err != nil {
+		return AssertError(fmt.Sprintf("Error loading UTXOSet: %s", err))
+	}
+
+	// Apply the stored tips to the virtual block.
+	tips := newBlockSet()
+	for _, tipHash := range dagState.TipHashes {
+		tip := dag.index.LookupNode(tipHash)
+		if tip == nil {
+			return AssertError(fmt.Sprintf("initDAGState: cannot find "+
+				"DAG tip %s in block index", dagState.TipHashes))
+		}
+		tips.add(tip)
+	}
+	dag.virtual.SetTips(tips)
+
+	// Set the last finality point
+	dag.lastFinalityPoint = dag.index.LookupNode(dagState.LastFinalityPoint)
+	dag.finalizeNodesBelowFinalityPoint(false)
+
+	// Go over any unprocessed blockNodes and process them now.
+	for _, node := range unprocessedBlockNodes {
+		// Check to see if the block exists in the block DB. If it
```

Comment:
Just a thought:
These are unprocessed block, theoretically we can boot and work without them.
So let us assume worst case: unprocessed blocks are corrupted AF.
In this case there can be two options:
A) Something else, more critical in the DB is corrupted AF as well.
B) Nothing else is corrupted.

In case A - we'll crash over `something else` pretty soon.
In case B - why not start working without the unprocessed blocks?

Reply:
P.S.
The other side of this argument is:
If something in the DB is corrupted, it might cause very unpredictable behavior in places that might be harder to detect, so we want to crush the moment we have a first indication of the crash.

I see the merit in both sides, and would love to hear your and @someone235 opinions on that.

Reply:
We spoke about it an decided to leave it as-is.